runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -5074,7 +5019,12 @@ static void jitStateLogic(J9JITConfig * jitConfig, TR::CompilationInfo * compInf
    if (IProfilerOffSinceStartup &&
        !TR::Options::getCmdLineOptions()->getOption(TR_DisableInterpreterProfiling) &&
        TR::Options::getCmdLineOptions()->getOption(TR_NoIProfilerDuringStartupPhase) &&
-       interpreterProfilingState == IPROFILING_STATE_OFF)
+       interpreterProfilingState == IPROFILING_STATE_OFF
+#if defined(J9VM_OPT_CRIU_SUPPORT)
```

Comment:
The way code is written, it will not take effect if the build has CRIU support but we are not in DebugOnRestore mode. Maybe it should be: 
&& (!jitConfig->javaVM->internalVMFunctions->isDebugOnRestoreEnabled(compInfo->getSamplerThread())
       || compInfo->getCRRuntime()->allowStateChange())
i.e. if debugOnRestore is enabled, do the same thing as we used to. Otherwise, consult `allowStateChange()`

Similar change is needed in different parts of this file.
Alternatively, you can hide `jitConfig->javaVM->internalVMFunctions->isDebugOnRestoreEnabled(compInfo->getSamplerThread()` check inside `allowStateChange()` and maybe cache it in a static.

Reply:
I had written that on purpose, since I only wanted this heuristic under debug on restore (given all the proactive compilations and what not that happen pre-checkpoint). However, I can apply this even for normal CRIU if you think it makes sense to.

Reply:
If debugOnRestore is not enabled, then the entire `if` statement is false. This means that we will never try to turn on IProfiler.

Reply:
Oh I see, yeah I'll have to fix this logic up.

Reply:
Fixed the logic in the last force push.